axonserver/src/main/java/io/axoniq/axonserver/access/jpa/User.java

```diff
@@ -108,4 +105,25 @@ public boolean equals(Object o) {
     public int hashCode() {
         return Objects.hash(userName);
     }
+
+    public static User newContextPermissions(User user) {
+        User newUser = new User(user.userName, user.password);
+        newUser.setRoles(user.getRoles()
+                             .stream()
+                             .filter(userRole -> "*".equals(userRole.getContext()))
+                             .collect(Collectors.toSet()));
+
+        return newUser;
+    }
+
+    public void removeContext(String context) {
+        for (Iterator<UserRole> contextIterator = roles.iterator(); contextIterator.hasNext(); ) {
+            UserRole userRole = contextIterator.next();
+            if (userRole.getContext().equals(context)) {
+                contextIterator.remove();
+                userRole.setUser(null);
```

Comment:
if the UserRole was a value object, we could simply remove it from the collection.

Reply:
done